cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -398,5 +456,14 @@ private void publishAudit(Id.DatasetInstance datasetInstance, AuditType auditTyp
     // TODO: Add properties to Audit Payload (CDAP-5220)
     AuditPublishers.publishAudit(auditPublisher, datasetInstance, auditType, AuditPayload.EMPTY_PAYLOAD);
   }
+
+  private void ensureAccess(DatasetId datasetId) throws Exception {
```

Comment:
the authorization APIs all throw ``Exception``, because authorization backends are extensions, and so they may have failures other than ``UnauthorizedException`` during each call. For example, the dataset backed authorizer may throw ``TransactionFailureException``. This was done so that they don't all have to wrap these other exceptions as unchecked exceptions.